src/com/vogon101/Graphics/Controls/ButtonControl.scala

```diff
@@ -0,0 +1,121 @@
+package com.vogon101.Graphics.Controls
+
+import com.colofabrix.scala.geometry.shapes.{Circle, Box}
+import com.colofabrix.scala.math.Vector2D
+import com.colofabrix.scala.tankwar.integration.TankEvaluator
+import org.lwjgl.opengl.GL11
+import org.newdawn.slick.opengl.Texture
+
+/**
+ * Created by Freddie on 18/05/2015.
+ *
+ * This class contains a Button. It has a render method, so is responsable for its own rendering
+ * as well as a Unit that it is it's OnClick action. Requires a Box as a renderable/clickable area and
+ * a Unit as an action
+ *
+ */
+class ButtonControl (_screenArea: Box, action : (Int) => Unit) {
+
+  //TODO: Make timing in seconds not steps
+  // ->TODO: Add a universal DeltaTime for the project
```

Comment:
Much better having a WorldRenderer linked to World that contains all the methods and the constants. Basically having separation between a world and a user interface.